lib/measures/tbd/resources/psi.rb

```diff
@@ -1154,6 +1153,43 @@ def processTBD(os_model, psi_set, io_path = nil, schema_path = nil)
     edge[:set] = p unless psi.empty?
   end                                                                # edge loop
 
+  # Add ":basement" or ":slab" keyword to edge if the following are met:
+  # 1. "foundation"-facing floor surface
+  # 2. 1x foundation-facing (:basement) or exterior-facing (:slab) wall
+  edges.values.each do |edge|
+    next unless gen_kiva
+    next if edge.has_key?(:basement)
```

Comment:
This is checking if the edge has already been processed?

Reply:
Yes.

Reply:
(this has been revised).